includes/class-edd-download.php

```diff
@@ -630,4 +641,31 @@ public function is_free( $price_id = false ) {
 
 	}
 
+	/**
+	 * Determine if the download supports a feature
+	 * 
+	 * @since  2.4
+	 * @param  string $feature   The feature being checked for support
+	 * @return bool              Whether the download supports the feature.
+	 */
+	public function supports( $features = '' ) {
+		$supports = apply_filters( 'edd_download_supports', $this->supports, $this->ID );
```

Comment:
It'd work the same way. It needs to be last so that extensions can see which features are already supported and then remove support for certain features if necessary too.